src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java

```diff
@@ -779,6 +782,89 @@ private float convertRowToArrow(
                 throw new SFException(ErrorCode.UNKNOWN_DATA_TYPE, logicalType, physicalType);
             }
             break;
+          case TIMESTAMP_TZ:
+            switch (physicalType) {
+              case SB8:
+                {
+                  StructVector structVector = (StructVector) vector;
+                  BigIntVector epochVector =
+                      (BigIntVector) structVector.getChild(FIELD_EPOCH_IN_SECONDS);
+                  IntVector timezoneVector = (IntVector) structVector.getChild(FIELD_TIME_ZONE);
+
+                  rowBufferSize += 0.25; // for children vector's null value
+                  structVector.setIndexDefined(curRowIndex);
+
+                  TimestampWrapper timestampWrapper =
+                      DataValidationUtil.validateAndParseTimestampTz(value, field.getMetadata());
+                  epochVector.setSafe(curRowIndex, timestampWrapper.getTimeInScale().longValue());
+                  timezoneVector.setSafe(
+                      curRowIndex,
+                      timestampWrapper
+                          .getTimeZoneIndex()
+                          .orElseThrow(
+                              () ->
+                                  new SFException(
+                                      ErrorCode.INVALID_ROW,
+                                      value,
+                                      "Unable to parse timezone for TIMESTAMP_TZ column")));
+                  rowBufferSize += 12;
```

Comment:
Where is 12 coming from?

Reply:
Summing the size of the [BigIntVector](https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/BigIntVector.html) (8) and the [IntVector](https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/IntVector.html) (4)